src/cat/cat.rs

```diff
@@ -276,18 +294,37 @@ fn open(path: &str) -> CatResult<InputHandle> {
 /// * `files` - There is no short circuit when encountiner an error
 /// reading a file in this vector
 fn write_fast(files: Vec<String>) -> CatResult<()> {
-    let mut writer = stdout();
     let mut in_buf = [0; 1024 * 64];
     let mut error_count = 0;
 
     for file in files {
         match open(&file[..]) {
-            Ok(mut handle) => while let Ok(n) = handle.reader.read(&mut in_buf) {
-                if n == 0 {
-                    break;
+            Ok(mut handle) => {
+                // If we're on Linux or Android, try to use the splice()
+                // system call for faster writing.
+                #[cfg(any(target_os = "linux", target_os = "android"))]
+                {
+                    match write_fast_with_splice(&mut handle) {
+                        Err(Sys(err)) => match err {
+                            EPIPE => {
```

Comment:
Rather than doing this, I think it makes more sense to do something like the following at the beginning of each program:
```rust
unsafe {
    libc::signal(libc::SIGPIPE, libc::SIG_DFL);
}
```

Currently, there's a [SIGPIPE handler for panics](https://github.com/uutils/coreutils/blob/036dd812958ace22d973acf7b370f58072049dac/src/uucore/panic.rs#L3) (installed in [mkmain.rs](https://github.com/uutils/coreutils/blob/036dd812958ace22d973acf7b370f58072049dac/mkmain.rs#L14) and [uutils.rs](https://github.com/uutils/coreutils/blob/036dd812958ace22d973acf7b370f58072049dac/src/uutils/uutils.rs#L39)), but that only occurs on `.unwrap()`, so it's probably just easier to change the signal back to its default.

I suppose we could also just ignore all `splice()`-related errors if we are falling back to the method using standard writes.

As a side note, if we keep the old `SIGPIPE` setup, we might need to change most `?` to `.unwrap()` for most read/write operations (as many utils just bubble the error up to main IIRC).  Alternatively, we could keep the current setup if we create some generic error type that gets bubbled up to the main function generated in `mkmain.rs`, which can then check for `SIGPIPE`.

Reply:
Thinking about this generally, I like the method where the error is bubbled up into the generated main function, but I'm not sure if it's possible to do so without allocating.

Reply:
Which approach do you think I should use?

Reply:
I think for now just ignoring errors with `splice` is fine (like, if there is an error, just use the default write rather than checking what error occurred).  In the future, I’d like to bubble up errors in all of the utilities though.